src/frontends/paddle/src/op/set_value.cpp

```diff
@@ -0,0 +1,157 @@
+    //    starts node: [1, 1, 1, 1, 1] -> [1, 1, 2, 1, 1].
+    // 4. Calculate and broadcast update_value to corresponding shape: [5, 6, 5, 3, 9].
+    //    1. Calculate `end - start`: [5, 3].
+    //    2. Calculate `(end - start) / step`: [2.5, 3]
+    //    3. Calculate `ceil((end - start) / step)`: [3, 3]
+    //    2. Use `ScatterNDUpdate` to get `target_value_shape` for input_shape: [5, 6, 7, 8, 9] -> [5, 6, 3, 3, 9].
+    //    3. Broadcast from [1, 6, 3, 3] to [5, 6, 3, 3, 9]
+    // 5. Create a range_node filled with number from 0 to numel.
+    //    1. Reshape it to input_shape.
+    //    2. Use `StridedSlice` to get those indices which are about to be updated.
+    // 6. Flatten input, update_value and sliced_range.
+    // 7. Use `ScatterUpdate` update update_value into input_data.
+    // 8. Reshape input to original input_shape.
+
+    const auto axes_node = default_opset::Constant::create(element::i64, {axes.size(), 1}, axes);
+    const auto spec_dim_node = std::make_shared<default_opset::GatherND>(input_shape, axes_node);
+    const auto zero_node = default_opset::Constant::create(element::i64, Shape{}, {0});
+    const auto one_node = default_opset::Constant::create(element::i64, Shape{}, {1});
+    const auto dim_node = default_opset::Constant::create(element::i64, Shape{}, {dims});
+    const auto reshape_flatten = default_opset::Constant::create(ov::element::i64, {1}, {-1});
+
+        // get positive starts ends and steps
+        if (node.has_input("StartsTensorList") && node.has_input("StepsTensorList") &&
+            node.has_input("EndsTensorList")) {
+        starts = handle_minus_index(node.get_ng_inputs("StartsTensorList"), spec_dim_node);
+        ends = handle_minus_index(node.get_ng_inputs("EndsTensorList"), spec_dim_node);
+        steps = handle_minus_index(node.get_ng_inputs("StepsTensorList"), spec_dim_node);
+    }
+    else if (node.has_attribute("starts") && node.has_attribute("steps") && node.has_attribute("ends")) {
+        starts = handle_minus_index(node.get_attribute<std::vector<int64_t>>("starts"), spec_dim_node);
+        ends = handle_minus_index(node.get_attribute<std::vector<int64_t>>("ends"), spec_dim_node);
+        auto step_vec = node.get_attribute<std::vector<int64_t>>("steps");
+        for (size_t i = 0; i < step_vec.size(); i++)
+            PADDLE_OP_CHECK(node, (step_vec[i] == 1), "Elements of steps must be 1");
+        steps = handle_minus_index(step_vec, spec_dim_node);
+    }
+    else PADDLE_OP_CHECK(node, (false), "Invalid arguments!");
+
+    // 3.1 get starts node
+    const auto zero_dim_node =
+        default_opset::Constant::create(element::i64, {static_cast<size_t>(dims)}, std::vector<int64_t>(dims));
+    starts_node = std::make_shared<default_opset::ScatterNDUpdate>(zero_dim_node, axes_node, starts);
+
+        // 3.2 get ends node
+        const auto ends_update_node = std::make_shared<default_opset::Subtract>(spec_dim_node, ends);
```

Comment:
Some lines are not aligned. Recommend to use build command 'make clang_format_fix_all' to automatically fix the code format issue.